### ISSUE TITLE
Awkward syntax when working with Promises

### ISSUE THREAD

**Issue by adrianmcli** [BuckleTypes/reason-js#12]
Currently, to fetch and log out an API response requires something like the following:

```rust
let p = ReasonJs.fetch "https://jsonplaceholder.typicode.com/posts/1";
ReasonJs.Promise.thenDo (ReasonJs.Promise.thenDo p ReasonJs.Response.text) Js.log;
```

This is rather unwieldy, but can be made somewhat cleaner by aliasing some of the functions:

```rust
let fetch = ReasonJs.fetch;
let then_ = ReasonJs.Promise.thenDo;
let p = ReasonJs.fetch "https://jsonplaceholder.typicode.com/posts/1";
then_ (then_ p ReasonJs.Response.text) Js.log;
```

By flipping the arguments, I am able to get an even nicer syntax:

```rust
let fetch = ReasonJs.fetch;
let then_ a b => ReasonJs.Promise.thenDo b a;
let p = fetch "https://jsonplaceholder.typicode.com/posts/1";
p |> then_ ReasonJs.Response.text |> then_ Js.log;
```

Perhaps the API can be reshaped to make usage more idiomatic.

**Comment by glennsl** [comment]
Fixed by #11, see https://github.com/BuckleTypes/reason-js/blob/master/src/core/promiseRe.re and the example https://github.com/BuckleTypes/reason-js/blob/master/examples/fetch_example.re

I plan on fleshing out both fetch and promise in the near future.

**Comment by adrianmcli** [comment]
@glennsl thanks for the reply. It looks like your PR has already been merged in, so I will close this issue.
